README.md

```diff
@@ -40,3 +40,42 @@ This repo contains a UML sequence diagram, which can be updated as follows:
 1. Run `docker pull think/plantuml` to download [this useful PlantUML image](https://hub.docker.com/r/think/plantuml/)
 2. Edit `./sequenceDiagram/source.uml` as needed
 3. Run `cat sequenceDiagram/source.uml | docker run --rm -i think/plantuml -tpng > sequenceDiagram/source.png` to output
+
+## Workflow
+
+The process of converting and sending a document is handled by a number of interrelated services:
+
+### Fetch task (run once a minute)
+
+- Fetch all new documents in the print queue from comino (compare with latest document in Dynamo)
+- Store in Dynamo (state = `Fetched`)
+- Push into SQS
+
+### Conversion task
+
+- Fetch document ID from SQS
+- Look up document in Comino and fetch from documents API
+- Convert RTF to PDF
+- Store converted document in S3
+- Update data in Dynamo (state = `Converted` / `Failed to convert`)
+
+### UI
+
+- Display documents in `Converted` state
+- User approves document to send
+- Update data in Dynamo (state = `Ready to send`)
+
+### Send to notify task (runs once per day at 3pm)
+
+- Fetch all documents from Dynamo that have state = `Ready to send`
+- Send to Notify
+  - If successful:
+    - Update Comino - remove from queue and update print date if successful
+    - Update data in Dynamo (state = `Sent to notify`)
+  - If unsuccessful:
+    - Update data in Dynamo (state = `Failed to send`)
+
+### Check notify task (runs once per hour)
+
+- Check state of documents with state `Sent to notify`
+- If Notify state has changed, update data in Dynamo (state = `Posted`)
```

Comment:
There's four possibilities of response from Gov Notify here  `GovNotifyPendingVirusCheck`, `GovNotifyVirusScanFailed`, `GovNotifyValidationFailed`, `LetterSent` . I think the unsuccessful bit from the point above will actually be caught in this task. I don't think notify just instantly sends back an unsuccessful response.